src/lib/storage/chunk.cpp

```diff
@@ -202,6 +202,7 @@ void Chunk::increase_invalid_row_count(const uint64_t count) const { _invalid_ro
 
 void Chunk::set_cleanup_commit_id(const CommitID cleanup_commit_id) {
   DebugAssert(!_cleanup_commit_id, "Cleanup commit ID can only be set once.");
+  std::unique_lock lock(_mutex_cleanup_commit_id);
```

Comment:
We can't run into any issues here, can we? The 64-bit value is guaranteed not to be read partially* (note that this is different from it being read atomically). A reader might see the old or the new version and either is fine. TSan doesn't know this, so using a suppression should be fine. Otherwise, at least use an atomic, which is significantly cheaper than a lock.

*) At least on Intel architectures.